lib/reek/context/code_context.rb

```diff
@@ -144,16 +143,35 @@ def config_for(detector_class)
           configuration_via_code_commment[detector_class.smell_type] || {})
       end
 
-      def track_visibility(visibility, names)
-        visibility_tracker.track_visibility children: children,
-                                            visibility: visibility,
-                                            names: names
-      end
-
       def number_of_statements
         statement_counter.value
       end
 
+      def module_context?
+        false
+      end
+
+      def method_context?
+        false
+      end
+
+      def singleton_method?
+        false
+      end
+
+      def instance_method?
+        false
+      end
+
+      def defined_as_instance_method?
+        false
+      end
+
+      def apply_current_visibility(current_visibility)
+        return unless defined_as_instance_method?
```

Comment:
MethodContexts and AttributContexts, yes. I'll see if I can push this down or into a module.